---
title: "Bird_tree_markdown"
output: html_document
date: "2023-04-25"
---

```{r}
#script for time calibration of bird phylogeny


library(picante)
library(ape)
library(stringr)
library(dplyr)
library(phytools) 
library(ggtree)

b_tree<- read.tree("Birds_7000.tre") 
tree_touse<- b_tree[[1]] #this is the maximum likelihood tre
labels<-tree_touse$tip.label

#char_tokeep should take 2 if we want family and 3 if we want genus and 4 if species
word_sep<- function(full_name, char_tokeep1, char_tokeep2 = char_tokeep1)
{
  if(grepl("-", full_name)){
    new_name_temp<- word(full_name,2, sep = "-")
    new_name<- word(new_name_temp, start = char_tokeep1, end = char_tokeep2,  sep = "_")
    return(new_name)
  }
  else{
    new_name<- word(full_name, start = char_tokeep1, end = char_tokeep2,  sep = "_")
    return(new_name)
  }
  
}


group_sep<-function(full_name)
{
  return(word(full_name, 1, sep = "-"))
}

group<- group_sep(labels)
genera<- unlist(lapply(labels, word_sep, 3))
species<- unlist(lapply(labels, word_sep, 3, 4))
families<- unlist(lapply(labels, word_sep, 2))

tree_touse$tip.label<- genera #now we have a tree of just genera tip labels. 

species_tree<- tree_touse
species_tree$tip.label <- species

write.tree(species_tree, "Bird_ML_700.tre") #write the maximum likelihood tree 


test_tree<- read.tree("Bird_ML_700.tre")

all_species<-test_tree$tip.label

#code for finding where the strigops species is 
grep("Strigops", labels)
labels[4564]

genus_tree<-tree_touse
genus_tree$tip.label <- genera

group_tree<- tree_touse
group_tree$tip.label<-group

family_tree<-tree_touse
family_tree$tip.label <- families
unique(group)




S2<- "Cariama"
name <- "Charadrius_vociferus"
name %in% species_tree$tip.label 


unique_genera<- sort(unique(genus_tree$tip.label))
unique_families<- sort(unique(family_tree$tip.label))
unique_groups<- sort(unique(group_tree$tip.label))
species_tree$tip.label



#now to get all the nodes for calibration
Calib1_node<- findMRCA(species_tree, c("Falco_peregrinus", "Cariama_cristata"))

Calib2_node<- findMRCA(species_tree, c("Alcippe_brunneicauda", "Pyrrhura_roseifrons"))

Calib3_node<- findMRCA(species_tree, c("Falco_peregrinus", "Microhierax_caerulescens"))

Calib4_node<- findMRCA(species_tree, c("Nestor_notabilis", "Strigops_habroptila"))

Calib6_node<- findMRCA(species_tree, c("Calyptomena_viridis", "Corvus_corax"))

Calib7_node<- findMRCA(species_tree, c("Menura_novaehollandiae", "Atrichornis_clamosus"))

Calib8_node<- findMRCA(species_tree, c("Melipotes_fumigatus", "Pardalotus_punctatus"))

Calib9_node<- findMRCA(species_tree, c("Orthonyx_temminckii", "Pomatostomus_temporalis"))

Calib10_node<- findMRCA(species_tree, c("Daphoenositta_chrysoptera", "Mohoua_albicilla"))

Calib11_node<- findMRCA(species_tree, c("Peltops_blainvillii",
"Strepera_graculina"))

Calib12_node<- findMRCA(species_tree, c("Ammodramus_savannarum", "Passerella_iliaca"))

Calib13_node<- findMRCA(species_tree, c("Regulus_calendula", "Certhia_americana"))


Jarvis2_node<- findMRCA(species_tree, c("Tinamus_guttatus", "Struthio_camelus"))

Jarvis4_node<- findMRCA(species_tree, c("Struthio_camelus", "Anas_platyrhynchos"))

Jarvis5_node<- findMRCA(species_tree, c("Anas_platyrhynchos", "Gallus_gallus"))


Jarvis5_node<- findMRCA(species_tree, c("Anas_platyrhynchos", "Gallus_gallus"))

Jarvis7_node<- findMRCA(species_tree, c("Mesitornis_unicolor", "Pterocles_gutturalis"))

Jarvis9_node<- findMRCA(species_tree, c("Orthonyx_temminckii", "Pomatostomus_temporalis"))

Jarvis12_node<- findMRCA(species_tree, c("Charadrius_vociferus", "Balearica_regulorum"))

#throws an error
#Jarvis14_node<- findMRCA(species_tree, c("Chlamydotis_macqueenii",  "Tauraco_erythrolophus"))


Jarvis21_node<- findMRCA(species_tree, c("Fulmarus_glacialis", "Aptenodytes_forsteri"))



Jarvis22_node<- findMRCA(species_tree, c("Pelecanus_crispus", "Nipponia_nippon"))

Jarvis23_node<- findMRCA(species_tree, c("Pelecanus_crispus", "Phalacrocorax_carbo"))


Jarvis29_node<- findMRCA(species_tree, c("Apaloderma_vittatum", "Buceros_rhinoceros"))


Jarvis31_node<- findMRCA(species_tree, c("Picoides_pubescens", "Buceros_rhinoceros"))


Jarvis35_node<- findMRCA(species_tree, c("Calypte_anna", "Chaetura_pelagica"))


Jarvis38_node<- findMRCA(species_tree, c("Manacus_vitellinus", "Corvus_brachyrhynchos"))


calibration_nodes<- c(Calib1_node, Calib2_node, Calib3_node, Calib4_node, Calib6_node, Calib7_node, Calib8_node, Calib9_node, Calib10_node, Calib11_node, Calib12_node, Calib13_node, Jarvis2_node, Jarvis4_node, Jarvis5_node, Jarvis7_node, Jarvis9_node, Jarvis12_node, Jarvis21_node, Jarvis21_node, Jarvis22_node, Jarvis23_node, Jarvis29_node, Jarvis31_node, Jarvis35_node, Jarvis38_node) 

ggtree(species_tree, layout = 'circular') + geom_point2(aes(subset = node%in% calibration_nodes),color = "green")


#steps
#Change the taxa names to only genus
#edit the tree and time calibrate based on genera 
#come back and change labels back to genus_species

#make a supplementary figure and circle each node (more of a visual check)
#cannot automate this unless we know for sure that topology supports the fossil assignment 
#want to make sure that the nodes are correct. 
#reduce to family level tree: can figure out what nodes should be constrained 
#numsites = number of genomic sites
  #the numsites: look at the burley tree and try to find how large the alignment is (how large the number of sites)
  #nucleotides used for the tree 
  #22 nuclear loci
    #number of sites is size of alignment 
    #in data matrix part of the methods
    # 24739240
#make a markdown file with citations and a bunch of sources. 
#smoothing parameter = rate smoothing estimation 
  #use default smoothing parameter
#mathematical problem of chronogram: how can we say that the branches are coming from a distribution of rates 
#that minimizes heterogenetiy of branches. 
#don't have to do anything too crazy: check topology of burley tree against what's published 
#just print out the jarvis trree phylogeny that shows where calibrations are and pnas tree (oliveras) and get 
#table of calibrations for two papers. Double check burley tree captures same nodes (probably does). Then assume that splits 
#are captured so that MRCA (give name corresponding with calibrations.)
#most have maximum age
#can run it with cross validation. better than arbitrary smoothing value. Need to run it both ways. 


#michael also talked to people he can contact about the +1 program 
print("no")

bird_tree_calib2<-read.tree("Bird_ML_700_dated2.tre")

plot(bird_tree_calib, show.tip.label = FALSE, type = "fan" )


bird_tree_calib4<- read.tree("Bird_ML_dated001_trial2.tre")
plot(bird_tree_calib4, type = "fan", show.tip.label = FALSE)


bird_tree_calib5<- read.tree("Bird_ML_0000000000000001_trial.tre")



ggtree(bird_tree_calib4, layout = 'circular') + geom_point2(aes(subset = node%in% calibration_nodes),color = "green")
ggsave("bird_tree_calibrated_with_nodes.png", dpi = 1200)


times_001<- branching.times(bird_tree_calib4)
times_0000001<- branching.times(bird_tree_calib5)
qqplot(times_0000001, times_001)


#now I will run the whole program on out_dates.tre 

#branching.times function for both trees, then do a qq plot 

times_noupper<-branching.times(bird_tree_calib3)
times_s1<- branching.times(bird_tree_calib)
times_slow<- branching.times(bird_tree_calib2)
times_slow_correct<- branching.times(bird_tree_calib4)

qqplot(times_slow_correct, times_slow)

bird_tree_calib<- bird_tree_calib4

#use the very small smoothing value. 
#produce a table of calibrated vs. actual values
#check to see if I calibrated the root node. 
#want hard upper bound on the ROOT

#produce a table: calibration vs value for calibrated node. pull out ages from same node from jarvis and oliveras trees. However this will be useful. 


```

## Notes for the calibration points

OLIVERAS


1. root of passerines? 

Calibration 1: Root (basal split in Australaves)
Minimum Age: 53.5 Ma
Maximum Age: 66.5 Ma
Taxa1: Falco peregrinus 
Taxa2: Cariama cristata 

2. split between Psittaciformes and Passeriformes
Minimum Age: 51.81 Ma
Maximum Age: 66.5 Ma
Taxa1: Alcippe_brunneicauda (passiformes)
Taxa2: Pyrrhura_roseifrons (psittaciformes)
Note: this is not based on the paper phylogeny. This is just based on random taxa chosen 

3. split between Falco and Microhierax
NOTE: no falco in the tree. 
Minimum Age: 16.0 Ma
Maximum Age: 48.0 Ma
Taxa1: Falco peregrinus 
Taxa2: Microhierax caerulescens


4. Crown Nestoridae (split between Strigops and Nestor) #for some reason Nestor isn't working. 
Minimum Age: 15.9 Ma
Maximum Age: 66.5 Ma
Taxa1:Nestor notabilis
Taxa2: Strigops habroptila
Note: Nestor does not exist in tree

5. Crown Acanthisittidae (split between Acanthisitta and Xenicus) #XENICUS not in tree. 
Minimum Age: 15.9 Ma
Maximum Age: 56.0 Ma
Taxa1:Acanthisitta chloris
Taxa2: 
Note: Neither genus exists in thee tree. 

6. Eupasseres (split between Suboscines and Oscines)
Minimum Age: 27.25 Ma
Maximum Age: 56.0 Ma
Taxa1: Calyptomena_viridis (suboscines)
Taxa2: Corvus_corax (oscines)


7. Split between Atrichornis and Menura
Minimum Age: 16.0 Ma
Maximum Age: 56.0 Ma
Taxa1: Menura_novaehollandiae (Menura)
Taxa2: Atrichornis clamosus (Atrichornithidae)

8. Split between Meliphagidae and Pardalotidae
Minimum Age: 14.22 Ma
Maximum Age: 56.0 Ma
Taxa1: Melipotes fumigatus (Meliphagidae)
Taxa2: Pardalotus punctatus (Paralotidae)


9.  Split between Orthonyx and Pomatostomus
Minimum Age: 17.72 Ma
Maximum Age: 56.0 Ma
Taxa1: Orthonyx temminckii
Taxa2: Pomatostomus temporalis

10. Stem Daphoenositta (split between Daphoenositta and Mohoua)
Minimum Age: 11.6 Ma
Maximum Age: 56.0 Ma
Taxa1: Daphoenositta chrysoptera
Taxa2: Mohoua albicilla

11. Crown Cracticidae (split between Peltops and (Cracticus + Strepera))
Minimum Age: 14.5 Ma
Maximum Age: 56.0 Ma
Taxa1: Peltops blainvillii
Taxa 2: Strepera graculina


12. stem Ammodramus (split between Ammodramus and other sparrows)
Minimum Age: 7.5 Ma
Maximum Age: 18.6 Ma
Taxa1: Ammodramus savannarum
Taxa2: Passerella iliaca


13. Split between Regulidae and Certhioidea
Minimum Age: 17.2 Ma
Maximum Age: 56.0 Ma
Taxa1: Regulus calendula
Taxa2: Certhia americana




#Not Oliveras (other one)

2) Terminal Branch "Struthio"
Minimum Age: 20 MY
Maximum Age: 20 MY? 
Taxa1: Tinamus_guttatus
Taxa2: Struthio_camelus

4)  Internal branch “Neognathae” (i.e., 46 bird genomes excluding Struthio+Tinamus) stem -Anseriformes
Minimum Age: 66 MY
Maximum Age: 70 MY
Taxa1: Struthio_camelus
Taxa2: Anas_platyrhynchos
NOTE: anas has some kind of subspecies, just took it out. 

5) Terminal Branch “Anas” = MRCA “Anas+Galliformes” Anseriformes, ?Anseranatidae
Minimum Age: 51 MY 
Maximum Age: 54 MY
Taxa1: Anas_platyrhynchos
Taxa2: Gallus_gallus

7) Terminal Branch "Pterocles" stem-Pterocliformes
Minimum Age: 24.7 MY
Maximum age: 24.7 MY?
Taxa1:Mesitornis_unicolor
Taxa2:Pterocles_gutturalis

9) Internal Branch “Phoenicopterus+Podiceps” 32 MY Grebe-Flamingo stem group (nocheck)
Minimum Age: 32 MY
Maximum age: 32 MY? 
Taxa1: Pterocles_gutturalis
Taxa2: Phoenicopterus_ruber

12) Terminal Branch “Balearica” 
Minimum Age: 32 MY
Maximum Age: 32 MY
Taxa1:Charadrius_vociferus
Taxa2: Balearica_regulorum

14) Terminal Branch “Turaco” ~32 MY (nocheck)
Minimum Age: 32 MY
Maxmimum Age: 32 MY
Taxa1: Chlamydotis_macqueenii
Taxa2: Tauraco_erythrolophus


21)Internal Branch “Pygoscelis+Aptenodytes”
Minimum Age: 60.5 MY
Maxmimum Age: 61.6 MY
Taxa1: Fulmarus_glacialis
Taxa2: Aptenodytes_forsteri

22)Internal Branch “Egretta+sister”
Minimum Age: 25.3 MY
Maximum Age; 26.39 MY
Taxa1: Pelecanus_crispus
Taxa2: Nipponia_nippon


23)Terminal Branch “Plegadis+sister”
Minimum Age: 47 MY
Maximum Age: 47 MY? 
Taxa1: Pelecanus_crispus
Taxa2: Phalacrocorax_carbo

24)Terminal Branch “Tyto” = MRCA “Tyto+sister”
Minimum Age: 56.8 MY
Maximum Age: 61.7 MY
Taxa1:Picoides_pubescens
Taxa2: Tyto_alba

29)Terminal Branch “Apaloderma” = MRCA “Apaloderma+sister” 
Minimum Age: 55.5 MY
Maxmimum Age: 55.5 MY
Taxa1: Apaloderma_vittatum
Taxa2: Buceros_rhinoceros

31) Terminal Branch: Buceros
Minimum Age: 47 MY
Maxmimum Age: 47 MY
Taxa1:Picoides_pubescens
Taxa2: Buceros_rhinoceros

34)Internal Branch “Chaetura+Calypte”
Minimum Age: 55.5 MY
Maxmimum Age: 55.5 MY
Taxa1: Calypte_anna
Taxa2: Antrostomus_carolinensis NOT IN TREE (don't use)

35) Terminal Branch “Calypte” 32 MY – MRCA hummingbird and swift
Minimum Age: 32 MY
Maxmimum Age: 32 MY
Taxa1: Calypte_anna
Taxa2: Chaetura_pelagica

38) Internal Branch “Manacus+Corvus+Taeniopygia+Geospiza”
Minimum Age: 13.6 MY
Maxmimum Age: 16.3 MY
Taxa1: Manacus_vitellinus
Taxa2: Corvus_brachyrhynchos

```{r}
install.packages("auk")
library(auk)
install.packages("readr")
library(readr)
library(ggtree)

cali_birds<-read_tsv(file = "true_cali_birds.csv")
cali_species<-cali_birds$species
cali_species<- gsub(" ", "_", cali_species)
cali_species<-data.frame(cali_species)
colnames(cali_species)<- "names"

remove_taxa<- function(species_list, master_phylogeny) #takes the output from check_taxa and removes them from the original dataframe of species
#removed this
{
  matched_species<-species_list%>% 
    filter(species_list$name%in%(master_phylogeny$tip.label))
  #return(matched_species)
}

#643 representative taxa out of 692. this is a good percentage, mark this down. 
species_tree<- remove_taxa(cali_species, bird_tree_calib)

sample_tree_generator<-function(sample, master_phylogeny) #generates a tree datatype from the cleaned sample data
{
  library(phytools)
  library(ape)
  list_species<- c(sample$name)
  sample_tree<-keep.tip(master_phylogeny, list_species) #maybe try to print this tree somehow later. 
  return(sample_tree)
}

cali_species_tree<- sample_tree_generator(species_tree, bird_tree_calib)

pd_app_picante(cali_species_tree, cali_species_tree)

plot(cali_species_tree, show.tip.label = FALSE )

ggtree(cali_species_tree, cex = 0.1, layout = "circular") +  theme_tree2()
print(635/689)
#92.1625% of california taxa are represented in the tree. 


#for the reserve systems:

reserve_birds<-read.csv(file = "uc_reserve_birds_inat.csv")
reserve_species<-unique(reserve_birds$taxon_name)
reserve_species<- gsub(" ", "_", reserve_species)
reserve_species<-data.frame(reserve_species)
colnames(reserve_species)<- "names"

reserve_taxa<-remove_taxa(reserve_species, bird_tree_calib)

#319 taxa observed in the UC reserves, 286 exist in the phylogeny

#286/319 (80.65%) 

reserve_species_tree<- sample_tree_generator(reserve_taxa, bird_tree_calib)
plot(reserve_species_tree, show.tip.label = FALSE)



##ggtree antics

nt<- ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% reserve_taxa$names), size=0.5, colour = 'red')


ggsave("reserve_on_cali.png", width=5, height=4, dpi=300, nt)
dev.off()

```
```{r}
#working with reserve data
#abbreviations


#AN = ana nuevo island reserve
#BX = Box Springs Reserve
#BPR = Burns Pinon Ridge Reserve
#CS = Carpenteria Salt Marsh Reserve
#DLM = Dawson Los Monos Canyon
#EC = Elliot Chaparral Reserve
#FO  = Fort Ord Natural Reserve
#HA = Hastings Reserve
#CAR = chickering American Reserve
#SJM = San Jacinto Mountains Reserve
#JP = Jepson Prairie Reserve
#KF = Kendall Frost Mission Reserve
#MV = Merced Vernal 
#MR = Motte Rimrock Reserve
#NRM = Norris Rancho Marino Reserve
#QR = Quail Ridge Reserve 
#SJ =  San Joaquin Marsh reserve
#SW = Sedgewick Reserve
#SGM = Sweeney Granite Mountains Reserve
#VA = Valentine aquatic Reserve
#VC = Valentine Camp reserve
#SC = Scripps Coastal Reserve
#SBA = Steele Burn Anza Borrego Reserve
# BD = Boyd Deep Canyon desert Reserve

#angelo coast range
BG<-read_tsv(file = "Bodega_reserve.csv")
BG_species<-BG$species
BG_df<- gsub(" ", "_", BG_species)
BG_df<-data.frame(BG_df)
colnames(BG_df)<- "names"
BG_taxa<- remove_taxa(BG_df, cali_species_tree)
BG_tree<-sample_tree_generator(BG_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% BG_taxa$names), size=0.7, colour = 'red')
write.tree(BG_tree, file = "bodega_tree.tre")


BX<-read_tsv(file = "Box_Springs_reserve.csv")
BX_species<-BX$species
BX_df<- gsub(" ", "_", BX_species)
BX_df<-data.frame(BX_df)
colnames(BX_df)<- "names"
BX_taxa<- remove_taxa(BX_df, cali_species_tree)
BX_tree<-sample_tree_generator(BX_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% BX_taxa$names), size=0.7, colour = 'red')
write.tree(BX_tree, file = "Box_Springs_tree.tre")

BPR<-read_tsv(file = "Burns_Pinon_Ridge_reserve.csv")
BPR_species<-BPR$species
BPR_df<- gsub(" ", "_", BPR_species)
BPR_df<-data.frame(BPR_df)
colnames(BPR_df)<- "names"
BPR_taxa<- remove_taxa(BPR_df, cali_species_tree)
BPR_tree<-sample_tree_generator(BPR_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% BPR_taxa$names), size=0.7, colour = 'red')
write.tree(BPR_tree, file = "Burns_Pinon_tree.tre")

BPR<-read_tsv(file = "Burns_Pinon_Ridge_reserve.csv")
BPR_species<-BPR$species
BPR_df<- gsub(" ", "_", BPR_species)
BPR_df<-data.frame(BPR_df)
colnames(BPR_df)<- "names"
BPR_taxa<- remove_taxa(BPR_df, cali_species_tree)
BPR_tree<-sample_tree_generator(BPR_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% BPR_taxa$names), size=0.7, colour = 'red')
write.tree(BPR_tree, file = "Burns_Pinon_tree.tre")


CS<-read_tsv(file = "Carpenteria_Salt_Marsh_reserve.csv")
CS_species<-CS$species
CS_df<- gsub(" ", "_", CS_species)
CS_df<-data.frame(CS_df)
colnames(CS_df)<- "names"
CS_taxa<- remove_taxa(CS_df, cali_species_tree)
CS_tree<-sample_tree_generator(CS_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% CS_taxa$names), size=0.7, colour = 'red')
ggsave("carpenteria_salt_vis.png", dpi = 1200)
write.tree(BPR_tree, file = "Carpenteria_Salt_tree.tre")

CAR<-read_tsv(file = "Chickering_American_River_Reserve.csv")
CAR_species<-CAR$species
CAR_df<- gsub(" ", "_", CAR_species)
CAR_df<-data.frame(CAR_df)
colnames(CAR_df)<- "names"
CAR_taxa<- remove_taxa(CAR_df, cali_species_tree)
CAR_tree<-sample_tree_generator(CAR_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% DLM_taxa$names), size=0.7, colour = 'red')
write.tree(CAR_tree, file = "Chickering_tree.tre")



DLM<-read_tsv(file = "Dawson_Los_Monos_Canyon_Reserve.csv")
DLM_species<-DLM$species
DLM_df<- gsub(" ", "_", DLM_species)
DLM_df<-data.frame(DLM_df)
colnames(DLM_df)<- "names"
DLM_taxa<- remove_taxa(DLM_df, cali_species_tree)
DLM_tree<-sample_tree_generator(DLM_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% DLM_taxa$names), size=0.7, colour = 'red')
write.tree(DLM_tree, file = "Dawson_Monos_tree.tre")



EC<-read_tsv(file = "Elliot_Chaparral_Reserve.csv")
EC_species<-EC$species
EC_df<- gsub(" ", "_", EC_species)
EC_df<-data.frame(EC_df)
colnames(EC_df)<- "names"
EC_taxa<- remove_taxa(EC_df, cali_species_tree)
EC_tree<-sample_tree_generator(EC_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% EC_taxa$names), size=0.7, colour = 'red')
write.tree(EC_tree, file = "Elliot_Chaparral_tree.tre")



FO<-read_tsv(file = "Fort_Ord_Natural_Reserve.csv")
FO_species<-FO$species
FO_df<- gsub(" ", "_", FO_species)
FO_df<-data.frame(FO_df)
colnames(FO_df)<- "names"
FO_taxa<- remove_taxa(FO_df, cali_species_tree)
FO_tree<-sample_tree_generator(FO_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% FO_taxa$names), size=0.7, colour = 'red')
write.tree(FO_tree, file = "Fort_Ord_tree.tre")


EO<-read_tsv(file = "Emerson_oaks_reserve.csv")
EO_species<-EO$species
EO_df<- gsub(" ", "_", EO_species)
EO_df<-data.frame(EO_df)
colnames(EO_df)<- "names"
EO_taxa<- remove_taxa(EO_df, cali_species_tree)
EO_tree<-sample_tree_generator(EO_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% EO_taxa$names), size=0.7, colour = 'red')
write.tree(EO_tree, file = "Emerson_oaks_tree.tre")


HA<-read_tsv(file = "Hastings_Reserve.csv")
HA_species<-HA$species
HA_df<- gsub(" ", "_", HA_species)
HA_df<-data.frame(HA_df)
colnames(HA_df)<- "names"
HA_taxa<- remove_taxa(HA_df, cali_species_tree)
HA_tree<-sample_tree_generator(HA_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% HA_taxa$names), size=0.7, colour = 'red')
write.tree(HA_tree, file = "Hastings_tree.tre")

SJM<-read_tsv(file = "San_Jacinto_Mountains_Reserve.csv")
SJM_species<-SJM$species
SJM_df<- gsub(" ", "_", SJM_species)
SJM_df<-data.frame(SJM_df)
colnames(SJM_df)<- "names"
SJM_taxa<- remove_taxa(SJM_df, cali_species_tree)
SJM_tree<-sample_tree_generator(SJM_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% SJM_taxa$names), size=0.7, colour = 'red')
write.tree(SJM_tree, file = "San_Jacinto_tree.tre")

JP<-read_tsv(file = "Jepson_Prairie_Reserve.csv")
JP_species<-JP$species
JP_df<- gsub(" ", "_", JP_species)
JP_df<-data.frame(JP_df)
colnames(JP_df)<- "names"
JP_taxa<- remove_taxa(JP_df, cali_species_tree)
JP_tree<-sample_tree_generator(JP_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% JP_taxa$names), size=0.7, colour = 'red')
ggsave("jepson_prairie_vis.png", dpi = 1200)
write.tree(SJM_tree, file = "Jepson_Prairie_tree.tre")


KF<-read_tsv(file = "Kendall_Frost_Mission_Bay_Marsh.csv")
KF_species<-KF$species
KF_df<- gsub(" ", "_", KF_species)
KF_df<-data.frame(KF_df)
colnames(KF_df)<- "names"
KF_taxa<- remove_taxa(KF_df, cali_species_tree)
KF_tree<-sample_tree_generator(KF_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% KF_taxa$names), size=0.7, colour = 'red')
write.tree(SJM_tree, file = "Kendall_Frost_tree.tre")

MV<-read_tsv(file = "Merced_Vernal_Pools_Reserve.csv")
MV_species<-MV$species
MV_df<- gsub(" ", "_", MV_species)
MV_df<-data.frame(MV_df)
colnames(MV_df)<- "names"
MV_taxa<- remove_taxa(MV_df, cali_species_tree)
MV_tree<-sample_tree_generator(MV_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% MV_taxa$names), size=0.7, colour = 'red')
write.tree(MV_tree, file = "Merced_Vernal_tree.tre")


MR<-read_tsv(file = "Motte_Rimrock_Reserve.csv")
MR_species<-MR$species
MR_df<- gsub(" ", "_", MR_species)
MR_df<-data.frame(MR_df)
colnames(MR_df)<- "names"
MR_taxa<- remove_taxa(MR_df, cali_species_tree)
MR_tree<-sample_tree_generator(MR_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% MR_taxa$names), size=0.7, colour = 'red')
write.tree(MR_tree, file = "Motte_Rimrock_Reserve.tre")


NRM<-read_tsv(file = "Norris_Rancho_Marino_Reserve.csv")
NRM_species<-NRM$species
NRM_df<- gsub(" ", "_", NRM_species)
NRM_df<-data.frame(NRM_df)
colnames(NRM_df)<- "names"
NRM_taxa<- remove_taxa(NRM_df, cali_species_tree)
NRM_tree<-sample_tree_generator(NRM_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% NRM_taxa$names), size=0.7, colour = 'red')
write.tree(NRM_tree, file = "Norris_Rancho_Marino_tree.tre")



QR<-read_tsv(file = "Quail_Ridge_Reserve.csv")
QR_species<-QR$species
QR_df<- gsub(" ", "_", QR_species)
QR_df<-data.frame(QR_df)
colnames(QR_df)<- "names"
QR_taxa<- remove_taxa(QR_df, cali_species_tree)
QR_tree<-sample_tree_generator(QR_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% QR_taxa$names), size=0.7, colour = 'red')
write.tree(QR_tree, file = "Quail_Ridge_tree.tre")



SJ<-read_tsv(file = "San_Joaquin_Marsh_Reserve.csv")
SJ_species<-SJ$species
SJ_df<- gsub(" ", "_", SJ_species)
SJ_df<-data.frame(SJ_df)
colnames(SJ_df)<- "names"
SJ_taxa<- remove_taxa(SJ_df, cali_species_tree)
SJ_tree<-sample_tree_generator(SJ_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% SJ_taxa$names), size=0.7, colour = 'red')
write.tree(SJ_tree, file = "San_Joaquin_Marsh_tree.tre")

SW<-read_tsv(file = "Sedgwick_Reserve.csv")
SW_species<-SW$species
SW_df<- gsub(" ", "_", SW_species)
SW_df<-data.frame(SW_df)
colnames(SW_df)<- "names"
SW_taxa<- remove_taxa(SW_df, cali_species_tree)
SW_tree<-sample_tree_generator(SW_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% SW_taxa$names), size=0.7, colour = 'red')
write.tree(SW_tree, file = "Sedgwick_Reserve_tree.tre")


SGM<-read_tsv(file = "Sweeney_Granite_Mountains_Reserve.csv")
SGM_species<-SGM$species
SGM_df<- gsub(" ", "_", SGM_species)
SGM_df<-data.frame(SGM_df)
colnames(SGM_df)<- "names"
SGM_taxa<- remove_taxa(SGM_df, cali_species_tree)
SGM_tree<-sample_tree_generator(SGM_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% SGM_taxa$names), size=0.7, colour = 'red')
write.tree(SGM_tree, file = "Sweeney_Granite_tree.tre")

VA<-read_tsv(file = "Valentine_Eastern_Sierra_Reserve_aquatic.csv")
VA_species<-VA$species
VA_df<- gsub(" ", "_", VA_species)
VA_df<-data.frame(VA_df)
colnames(VA_df)<- "names"
VA_taxa<- remove_taxa(VA_df, cali_species_tree)
VA_tree<-sample_tree_generator(VA_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% VA_taxa$names), size=0.7, colour = 'red')
write.tree(VA_tree, file = "Valentine_aquatic_tree.tre")

VC<-read_tsv(file = "Valentine_Eastern_Sierra_Reserve_Camp.csv")
VC_species<-VC$species
VC_df<- gsub(" ", "_", VC_species)
VC_df<-data.frame(VC_df)
colnames(VC_df)<- "names"
VC_taxa<- remove_taxa(VC_df, cali_species_tree)
VC_tree<-sample_tree_generator(VC_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% VC_taxa$names), size=0.7, colour = 'red')
write.tree(VC_tree, file = "Valentine_camp_tree.tre")

SC<-read_tsv(file = "Scripps_Coastal_Reserve.csv")
SC_species<-SC$species
SC_df<- gsub(" ", "_", SC_species)
SC_df<-data.frame(SC_df)
colnames(SC_df)<- "names"
SC_taxa<- remove_taxa(SC_df, cali_species_tree)
SC_tree<-sample_tree_generator(SC_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% SC_taxa$names), size=0.7, colour = 'red')
write.tree(SC_tree, file = "Scripps_Coastal_tree.tre")

SBA<-read_tsv(file = "Steele_Burn_Anza_Borrego_Research_Center.csv")
SBA_species<-SBA$species
SBA_df<- gsub(" ", "_", SBA_species)
SBA_df<-data.frame(SBA_df)
colnames(SBA_df)<- "names"
SBA_taxa<- remove_taxa(SBA_df, cali_species_tree)
SBA_tree<-sample_tree_generator(SBA_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% SBA_taxa$names), size=0.7, colour = 'red')
write.tree(SBA_tree, file = "Steele_Burn_Anza_Borrego_tree.tre")

BD<-read_tsv(file = "Boyd_Deep_Canyon_desert_Reserve.csv")
BD_species<-BD$species
BD_df<- gsub(" ", "_", BD_species)
BD_df<-data.frame(BD_df)
colnames(BD_df)<- "names"
BD_taxa<- remove_taxa(BD_df, cali_species_tree)
BD_tree<-sample_tree_generator(BD_taxa,cali_species_tree)
ggtree(cali_species_tree, layout = "fan") +
    geom_tiplab(aes(subset = label%in% BD_taxa$names), size=0.7, colour = 'red')
write.tree(BD_tree, file = "Boyd_Deep_Canyon_tree.tre")



commdata_maker<- function(tree)
{
  comm<-data.frame(row.names = tree$tip.label, clump1 = 1, clump2 = tree$tip.label)
  comm2<-subset(comm, select = clump1)
  return(t(as.matrix(comm2)))
}

pd(commdata_maker(tree_random520), tipShuffle(cali_species_tree))

pd_CSM<- pd_app_picante(CS_tree, cali_species_tree)
pd_full<-pd_app_picante(cali_species_tree, tipShuffle(cali_species_tree))


tree_random80<-read.tree("sample_trees_birds/sample_tree_cluster80")
pd_random_sample<- pd_app_picante(tree_random80, tipShuffle(cali_species_tree))

```



```{r}
pd_app_picante<-function(sample_tree, master_phylogeny)
{
  comm_sample<-data.frame(row.names = sample_tree$tip.label, clump1 = 1, clump2 = sample_tree$tip.label)
  comm_sample_touse<-t(as.matrix(subset(comm_sample, select = clump1)))
  pd_val<- picante::pd(comm_sample_touse,master_phylogeny)
  return(pd_val)
}


mpd_app_picante<-function(sample_tree,coph_mat)
{
  comm_sample<-data.frame(row.names = sample_tree$tip.label, clump1 = 1, clump2 = sample_tree$tip.label)
  comm_sample_touse<-t(as.matrix(subset(comm_sample, select = clump1)))
  mpd_val<- mpd(comm_sample_touse,coph_mat)
  return(mpd_val)
}

mntd_app_picante<-function(sample_tree,coph_mat)
{
  comm_sample<-data.frame(row.names = sample_tree$tip.label, clump1 = 1, clump2 = sample_tree$tip.label)
  comm_sample_touse<-t(as.matrix(subset(comm_sample, select = clump1)))
  mntd_val<- mntd(comm_sample_touse,coph_mat)
  return(mntd_val)
}

pd_cali<-pd_app_picante(cali_species_tree, bird_tree_calib)
pd_reserves<-pd_app_picante(reserve_species_tree, bird_tree_calib)

```
```{r}
#this is for PD analysis

library(data.table)
library(drc)

#define tree sizes (this is for birds now)
tree_sizes= seq(5, 430, by=5) #this is a list of tree sizes. 

 
mntd_data<-read.csv("CI_mntd_output_bootstrap_bird.csv")
mpd_data<-read.csv("CI_mpd_output_bootstrap_bird.csv")
pd_data<-read.csv("CI_pd_output_bootstrap_bird.csv")

#for mpd
mpd_low <- transpose(mpd_data[1,])
mpd_med <- transpose(mpd_data[3,])
mpd_high<- transpose(mpd_data[2,])
#tree sizes from above. reduce this redundancy.

mpd_low <- data.frame(mpd_low[-1,])
mpd_med<-data.frame(mpd_med[-1,])
mpd_high<-data.frame(mpd_high[-1,])
colnames(mpd_low) <- c("mpd")
colnames(mpd_med)<-c("mpd")
colnames(mpd_high)<-c("mpd")
mpd_low$mpd = as.double(mpd_low$mpd)
mpd_med$mpd = as.double(mpd_med$mpd)
mpd_high$mpd = as.double(mpd_high$mpd)
mpd_low["sizes"] = tree_sizes
mpd_med["sizes"] = tree_sizes
mpd_high["sizes"] = tree_sizes

#for mntd
mntd_low <- transpose(mntd_data[1,])
mntd_med <- transpose(mntd_data[3,])
mntd_high<- transpose(mntd_data[2,])
#tree sizes from above. reduce this redundancy.

mntd_low <- data.frame(mntd_low[-1,])
mntd_med<-data.frame(mntd_med[-1,])
mntd_high<-data.frame(mntd_high[-1,])
colnames(mntd_low) <- c("mntd")
colnames(mntd_med)<-c("mntd")
colnames(mntd_high)<-c("mntd")
mntd_low$mntd = as.double(mntd_low$mntd)
mntd_med$mntd = as.double(mntd_med$mntd)
mntd_high$mntd = as.double(mntd_high$mntd)
mntd_low["sizes"] = tree_sizes
mntd_med["sizes"] = tree_sizes
mntd_high["sizes"] = tree_sizes

plot(mntd_low$mntd~mntd_low$sizes)
#for pd
pd_low <- transpose(pd_data[1,])
pd_med <- transpose(pd_data[3,])
pd_high<- transpose(pd_data[2,])
#tree sizes from above. reduce this redundancy.

pd_low <- data.frame(pd_low[-1,])
pd_med<-data.frame(pd_med[-1,])
pd_high<-data.frame(pd_high[-1,])
colnames(pd_low) <- c("pd")
colnames(pd_med)<-c("pd")
colnames(pd_high)<-c("pd")
pd_low$pd = as.double(pd_low$pd)
pd_med$pd = as.double(pd_med$pd)
pd_high$pd = as.double(pd_high$pd)
pd_low["sizes"] = tree_sizes
pd_med["sizes"] = tree_sizes
pd_high["sizes"] = tree_sizes


#model fitting
#mpd

model_low <- drm(mpd_low$mpd ~ tree_sizes, fct = MM.3())
model_med <- drm(mpd_med$mpd ~ tree_sizes, fct = MM.3())
model_high <- drm(mpd_high$mpd ~ tree_sizes, fct = MM.3())

plot(model_low)+
plot(model_high)+
plot(model_med)

summary(model_low)
summary(model_med)
summary(model_high) 



# mntd. need to work on these fits 

library(stats)
library(ggplot2)
library(broom)


#attempt 1: negative exponential
mntd_fit_equation<- y0 + (yf - y0) * exp(-exp(log_alpha) * (x - tree_sizes))


fit_low <- nls(mntd ~ SSasymp(tree_sizes, yf, y0, log_alpha), data = mntd_low)
fit_med<- nls(mntd ~ SSasymp(tree_sizes, yf, y0, log_alpha), data = mntd_med)
fit_high<- nls(mntd ~ SSasymp(tree_sizes, yf, y0, log_alpha), data = mntd_high)


augmented_data <- augment(fit_med)

ggplot(augmented_data, aes(x = tree_sizes, y = mntd)) +
  geom_point() +
  geom_line(aes(y = .fitted))


#attempt 2: michaelis menten 
mntd_low_fit <- drm(mntd_low$mntd ~ tree_sizes, fct = MM.3())
mntd_med_fit <- drm(mntd_med$mntd ~ tree_sizes, fct = MM.3())
mntd_high_fit <- drm(mntd_high$mntd ~ tree_sizes, fct = MM.3())

plot(mntd_high)+ points(mntd_med) + points(mntd_low)
plot(mntd_high)+
plot(mntd_med)

summary(mntd_low_fit)
summary(mntd_high_fit)
summary(mntd_med_fit)



#for pd

```

```{r}

#functions for predicting the mpd. Also work for mntd. 
range_pred<- function(sample_size, model_low, model_mean, model_high) #need to connect this to the models. This predicts the lower, mean and upper range for a given sample size. 
{
  
  lower <- model_low$coefficients[1] + (model_low$coefficients[2] -model_low$coefficients[1])/(1+ model_low$coefficients[3]/sample_size)
  mean <- model_mean$coefficients[1] + (model_mean$coefficients[2] -model_mean$coefficients[1])/(1+ model_mean$coefficients[3]/sample_size)
  upper <- model_high$coefficients[1] + (model_high$coefficients[2] -model_high$coefficients[1])/(1+ model_high$coefficients[3]/sample_size)
  return(c(lower[[1]], mean[[1]], upper[[1]]))
}


 
is_significant<-function(range_prediction, true_pd) #boolean function for whether the observed pd lies within the 95% CI
{
  low_lim = range_prediction[1]
  upp_lim = range_prediction[3]
  return(!(between(true_pd, low_lim, upp_lim))) #returns false if not significant and true if significant (false if falls in range. )
}


#for predicting mntd (michaelis menten also works )


```

another part
```{r}



#comparing the statistics to the hastings reserve stuff. 
#need to make a cophenetic matrix. 
#list of hasting reserve trees
#cophen_touse has been temporarily defined. neneds to be less hacky. 


#KF = Kendall Frost Mission Reserve
#MV = Merced Vernal 
#MR = Motte Rimrock Reserve
#NRM = Norris Rancho Marino Reserve
#QR = Quail Ridge Reserve 
#SJ =  San Joaquin Marsh reserve
#SW = Sedgewick Reserve
#SGM = Sweeney Granite Mountains Reserve
#VA = Valentine aquatic Reserve
#VC = Valentine Camp reserve
#SC = Scripps Coastal Reserve
#SBA = Steele Burn Anza Borrego Reserve
# BD = Boyd Deep Canyon desert Reserve

trees_list<- c(BX_tree, BPR_tree, CS_tree, DLM_tree, EC_tree, FO_tree, HA_tree, CAR_tree, SJM_tree, JP_tree, KF_tree, MV_tree, MR_tree, NRM_tree, QR_tree, SJ_tree, SW_tree, SGM_tree, VA_tree, VC_tree, SC_tree, SBA_tree, BD_tree)



mpd_vals<-rep(NA, length(trees_list))
pd_vals<-rep(NA, length(trees_list))
mntd_vals<-rep(NA, length(trees_list))
family_sizes<-rep(NA, length(trees_list))
mpd_significant<-rep(NA, length(trees_list))
mntd_significant<-rep(NA, length(trees_list))

for(i in 1:length(trees_list))
{
  phylo_temp<- trees_list[[i]]
  if(length(phylo_temp$tip.label) <=1)
  {
   pd_vals[i]<-NA
   mpd_vals[i]<-NA
   mntd_vals[i]<-NA
   family_sizes[i]<-1
  }
  else
  {
  pd_vals[i]<-pd_app_picante(phylo_temp, cali_bird_phylo)$PD
  mpd_vals[i]<-mpd_app_picante(phylo_temp, cophen_touse)
  mntd_vals[i]<-mntd_app_picante(phylo_temp, cophen_touse)
  family_sizes[i]<-length(phylo_temp$tip.label)
  mpd_significant[i]<-is_significant(range_pred(family_sizes[i], model_low, model_med, model_high), mpd_vals[i])
  mntd_significant[i]<- is_significant(range_pred(family_sizes[i], mntd_low_fit, mntd_med_fit, mntd_high_fit), mntd_vals[i])
  }
}

rnames<- c(  "Box Springs Reserve", "Burns Pinon Ridge Reserve", "Carpenteria Salt Marsh Reserve", "Dawson Los Monos Canyon Reserve", "Elliot Chaparral Reserve", "Fort Ord Natural Reserve", "Hastings Reserve", "chickering American Reserve", "San Jacinto Mountains Reserve" , "Jepson Prairie Reserve, ", " Kendall Frost Mission Reserve", "Merced VernalPool",  " Motte Rimrock Reserve", "Norris Rancho Marino Reserve", "Quail Ridge Reserve", "San Joaquin Marsh reserve", "Sedgewick Reserve", "Sweeney Granite Mountains Reserve", "Valentine aquatic Reserve", "Valentine Camp reserve", "Scripps Coastal Reserve", "Steele Burn Anza Borrego Reserve", "Boyd Deep Canyon desert Reserve")

bird_statistics<-data.frame(mpd_vals)
bird_statistics$pd_vals<-pd_vals
bird_statistics$mntd_vals<-mntd_vals
bird_statistics$family_sizes<-family_sizes
bird_statistics$mpd_significance<- mpd_significant
bird_statistics$mntd_significance<- mntd_significant
rownames(bird_statistics)<-rnames


#for mpd
x<- seq(0,250)
ylow<- model_low$coefficients[1] + (model_low$coefficients[2] -model_low$coefficients[1])/(1+ model_low$coefficients[3]/x)
yhigh<- model_high$coefficients[1] + (model_high$coefficients[2] -model_high$coefficients[1])/(1+ model_high$coefficients[3]/x)
plot(x, ylow, type = "l", ylab = "mpd", ylim = c(10,130), xlab = "tree size")
lines(x, yhigh, type = "l", col = 2)
polygon(c(x, rev(x)), c(ylow, rev(yhigh)),
        col = "#6BD7AF")

points(bird_statistics$family_sizes, bird_statistics$mpd_vals, cex = 0.5)
#text(bird_statistics$family_size, bird_statistics$mpd_vals + 5, labels= c( "BX", "BPR", "CS", "DLM", "EX", "FO", "HA", "CAR", "SJM", "JP")

#for mntd
x<- seq(0,250)
ylow<- mntd_low_fit$coefficients[1] + (mntd_low_fit$coefficients[2] -mntd_low_fit$coefficients[1])/(1+ mntd_low_fit$coefficients[3]/x)
yhigh<- mntd_high_fit$coefficients[1] + (mntd_high_fit$coefficients[2] -mntd_high_fit$coefficients[1])/(1+ mntd_high_fit$coefficients[3]/x)
plot(x, ylow, type = "l", ylab = "mpd", ylim = c(0,145), xlab = "tree size")
lines(x, yhigh, type = "l", col = 2)
polygon(c(x, rev(x)), c(ylow, rev(yhigh)),
        col = "#6BD7AF")
points(bird_statistics$family_sizes, bird_statistics$mntd_vals, cex = 0.5)
points(mntd_low$mntd~mntd_low$sizes, cex = 0.1)
points(mntd_med$mntd~mntd_med$sizes, cex = 0.1)
points(mntd_high$mntd~mntd_high$sizes, cex = 0.1)



#saving table as image


#here is pd

plot(pd_low$pd~ pd_low$sizes, type = "l")
lines(pd_high$pd ~pd_high$sizes, type = "l")
points(bird_statistics$pd_vals~ bird_statistics$family_sizes)


```


